src/controller/SetUpCodePairer.cpp

```diff
@@ -62,12 +66,18 @@ CHIP_ERROR SetUpCodePairer::PairDevice(NodeId remoteId, const char * setUpCode,
 
     ResetDiscoveryState();
 
-    ReturnErrorOnFailure(Connect(payload));
+    if (resolutionData.HasValue())
+    {
+        ReturnErrorOnFailure(mSystemLayer->StartTimer(System::Clock::Milliseconds32(kDeviceDiscoveredTimeout),
```

Comment:
Probably worth documenting why we are starting the timer here even though we are not doing any discovery to time out.  Because it's not clear to me why we are.

Reply:
I have removed it.